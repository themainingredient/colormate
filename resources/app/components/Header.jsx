import React from 'react';
import styled from 'styled-components';
import Globals, { flexCenter } from '../Global.styles';

import ColormateLogo from '../assets/colormateLogo.svg';

const { colors, fonts } = Globals;

// TODO: Put recurring css in global file
const HeaderWrapper = styled.div`
  ${flexCenter};
  height: 87px;
  width: 320px;
  background-color: ${colors.LightGrey};
`;

const Tag = styled.div`
  height: 22px;
  width: 39px;
  background-color: ${colors.TMIBlue};
  border-radius: 11px;
  position: absolute;
  top: 8px;
  right: 8px;
  color: ${colors.White};
  font-family: ${fonts.SFPro.reg};
  font-size: 11px;
  line-height: 24px;
  text-align: center;
`;

const Header = () => (
  <HeaderWrapper>
    <ColormateLogo height={66} width={186} />
    <Tag>v0.1</Tag>
  </HeaderWrapper>
);

export default Header;
