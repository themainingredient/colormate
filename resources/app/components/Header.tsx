import React from 'react';
import styled from 'styled-components';
import Globals, { flexCenter } from '../Global.styles';

import ColormateLogo from '../assets/colormateLogo.svg';

const { colors, fonts } = Globals;
const isBeta = process.env.REACT_APP_IS_BETA;
const VERSION = isBeta ? `${process.env.REACT_APP_VERSION}-beta` : process.env.REACT_APP_VERSION;

const HeaderWrapper = styled.div`
  ${flexCenter};
  height: 87px;
  width: 100%;
  background-color: ${colors.LightGrey};
  box-shadow: 0 5px 10px 2px ${colors.Black25};
`;

const Tag = styled.div`
  height: 22px;
  width: ${isBeta ? '60px' : '39px'};
  background-color: ${colors.TMIBlue};
  border-radius: 11px;
  position: absolute;
  top: 8px;
  right: 8px;
  color: ${colors.White};
  font-family: ${fonts.SFPro.reg};
  font-size: 11px;
  line-height: 24px;
  text-align: center;
`;

const StyledColormateLogo = styled(ColormateLogo)`
  position: absolute;
  top: 50%;
  left: 47%;
  transform: translate(-50%, -50%);
`;

const Header = () => (
  <HeaderWrapper>
    <StyledColormateLogo />
    <Tag>{VERSION}</Tag>
  </HeaderWrapper>
);

export default Header;
