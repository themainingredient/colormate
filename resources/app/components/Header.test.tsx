import React from 'react';
import { render, cleanup } from 'react-testing-library';
import Header from './Header';

jest.mock('../assets/colormateLogo.svg', () => 'img');

afterEach(cleanup);

describe('Header.jsx', () => {
  test('renders and displays correctly', () => {
    const { container } = render(<Header />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
