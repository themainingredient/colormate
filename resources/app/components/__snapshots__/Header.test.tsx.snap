// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.jsx renders and displays correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 87px;
  width: 100%;
  background-color: #E8E9EF;
  box-shadow: 0 5px 10px 2px #00000040;
}

.c2 {
  height: 22px;
  width: 39px;
  background-color: #4E41FF;
  border-radius: 11px;
  position: absolute;
  top: 8px;
  right: 8px;
  color: #FFFFFF;
  font-family: SFProRegular;
  font-size: 11px;
  line-height: 24px;
  text-align: center;
}

.c1 {
  position: absolute;
  top: 50%;
  left: 47%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

<div
  class="c0"
>
  <img
    class="c1"
  />
  <div
    class="c2"
  />
</div>
`;
